opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -232,61 +284,246 @@ message MetricDescriptor {
-
-  // start_time_unix_nano is the time when the cumulative value was reset to zero.
-  // This is used for Counter type only. For Gauge the value is not specified and
-  // defaults to 0.
+  // Kind explains how the DataPoint was produced (called
+  // "Structure"), how the point was aggregated with resepect to time
+  // (called "Temporality"), whether it was computed asynchronously,
+  // and (when the Structure is ADDING) whether the sum is not
+  // monotonic.
   //
-  // The cumulative value is over the time interval (start_time_unix_nano, time_unix_nano].
-  // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+  // Kind names are generated from valid combinations of KindElement
+  // by joining the effective KindElements using underscores.  There
+  // are:
   //
-  // Value of 0 indicates that the timestamp is unspecified. In that case the timestamp
-  // may be decided by the backend.
-  fixed64 start_time_unix_nano = 2;
-
-  // time_unix_nano is the moment when this value was recorded.
-  // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
-  fixed64 time_unix_nano = 3;
+  // - 3 possibilities for Temporality
+  // - 3 possibilities for Structure/Monotonicity: Adding Monotonic,
+  //     Adding Non-Monotonic, and Grouping
+  // - 2 possibilities of being Asynchronous or not.
+  //
+  // Exclusing the asynchronous instantaneous combinations, this makes
+  // 15 valid values.
+  //
+  // The cases are detailed below in terms of:
+  //
+  // - How StartTimeUnixNano and TimeUnixNano are interpreted?
+  // - Which OpenTelemetry is likely to produce instrument?
+  // - Under which typical aggregations?
+  // - Producing which kind of data point?
+  //
+  // Certain general statements can be made.
+  //
+  // About the two Structure kinds:
+  //
+  // 1. ADDING kinds associate with (UpDown)Counter and (UpDown)SumObserver instruments
+  //    - MONOTONIC kinds associate with Counter and SumObserver
+  //    - non-MONOTONIC kinds associate with UpDownCounter and UpDownSumObserver
+  // 2. GROUPING kinds associate with ValueRecorder and ValueObserver instruments
+  // 3. GROUPING and ADDING kinds both typically yield Histogram data points.
+  //
+  // About ASYNCHRONOUS kinds:
+  // 
+  // 1. ASYNCHRONOUS kinds associate with Observer instruments
+  // 2. INSTANTANEOUS temporality does not combine with ASYNCHRONOUS kinds.
+  //
+  // About SCALAR_INT64 and SCALAR_DOUBLE value type (TODO(#159) and
+  // equally the RAW value types):
+  //
+  // 1. SCALAR value type interpretation depends on
+  //    both temporality and structure
+  // 2. For GROUPING kinds:
+  //    - For INSTANTANEOUS temporality: An individual measurement
+  //    - For DELTA or CUMULATIVE temporality: The last measured individual value
+  //      in the collection interval.
+  // 3. For ADDING kinds:
+  //    - For INSTANTANEOUS temporality: An instantaneously measured change in a sum
```

Comment:
You lost me. I get the need for including if a sum is a delta or cumulative but `INSTANTANEOUS`?

Reply:
This comment tells me you're not lost. :grinning:

There is something here, and I considered making an argument that ADDING/INSTANTANEOUS is either equivalent to ADDING/DELTA or to ADDING/CUMULATIVE, so why include this possibility?

I reasoned that ADDING/INSTANTANEOUS means "adding without a start time", meaning the ADDING/DELTA interpretation is the only good interpretation. 

I would consider adding more example to help here. The interpretation is:
- ADDING/INSTANTANEOUS an individual DELTA event as might emit from a stateless counter. There is no StartTimeUnixNano set in this case (and no synchronization required)
- ADDING/DELTA is a sum of 1 or more instantaneous events with StartTimeUnixNano set.
- ADDING/CUMULATIVE is a sum of 1 or more instantaneous events with StartTimeUnixNano set.

Reply:
I added
```
  // About ADDING|INSTANTANEOUS vs ADDING|DELTA:
  //
  // 1. An INSTANTANEOUS|ADDING event is stateless, has no StartTimeUnixNano.
  // 2. A DELTA|ADDING event remembers its start time, sets StartTimeUnixNano.
  //
```